validphys2/src/validphys/process_options.py

```diff
@@ -189,6 +189,17 @@ def _displusjet_xq2map(kin_dict):
     return x, q2
 
 
+def _dptboson_xq2map(kin_dict):
+    """Compute x and q2 mapping for DY Z -> ll + jet process.
+    Here pT refers to the transverse momentum of the Z boson."""
+    pT = kin_dict[_Vars.pT]
+    m_Z2 = kin_dict[_Vars.m_Z2]
+    sqrts = kin_dict[_Vars.sqrts]
+    x = np.sqrt(m_Z2 + pT * pT) / sqrts
+    q2 = m_Z2 + pT * pT
+    return x, q2
```

Comment:
I took [kintransforms.py](https://github.com/NNPDF/nnpdf/blob/master/validphys2/src/validphys/plotoptions/kintransforms.py) as a reference for the map. Is that correct?

Reply:
There were some bugs there, you can try creating the x-q2 plot https://vp.nnpdf.science/vxS8mWGySsu-cehRFnhSOA==/ to see how it looks for these datasets.

Reply:
This is what I get: https://vp.nnpdf.science/y1HWyAGhT0Sm1rubBeO5Xw==